pkg/glusterfs/controllerserver.go

```diff
@@ -56,16 +97,21 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 	volSizeMB := int(utils.RoundUpSize(volSizeBytes, 1024*1024))
 
-	// Get Volume name : TODO use the values from request
-	volumeName := req.Name
-	glusterVol := req.GetParameters()["glustervol"]
-	glusterServer = req.GetParameters()["glusterserver"]
-	glusterURL := req.GetParameters()["glusterurl"]
-	glusterURLPort := req.GetParameters()["glusterurlport"]
-	glusterUser := req.GetParameters()["glusteruser"]
-	glusterUserSecret := req.GetParameters()["glusterusersecret"]
+	// parse the request.
+	parseResp, parseErr := cs.ParseRequest(req)
+	if parseErr != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to parse request")
+	}
+
+	if parseResp != nil && parseResp.gdVolReq != nil {
```

Comment:
same here as well, as we are not returning nil here, we don't need this one also

Reply:
@Madhu-1 same as above.